### PR TITLE
Pt 163083413 sc signatures check

### PR DESCRIPTION
**Pull request by velzevur**
PTs:
* [SC FSM tests: reusable ports](https://www.pivotaltracker.com/story/show/159456827)
* [Add SC off-chain signatures validation](https://www.pivotaltracker.com/story/show/163083413)

### CODE REVIEW

**Comment by velzevur** [comment]
Added a `WIP` as it must be based on @uwiger 's renaming changes

**Comment by velzevur** [comment]
It must be based on #2048 and accomodate the solo closing sequence

**Comment by velzevur** [comment]
Squashed and rebased on latest `master`. The solo closing sequence is not in place yet, so no accomodation regarding it.

File: apps/aechannel/src/aesc_fsm.erl [hunk 1]
```diff
@@ -1264,6 +1340,8 @@ handle_call_(St, close_solo, From, D) ->
             D1 = request_signing(close_solo_tx, CloseSoloTx, D),
             next_state(awaiting_signature, D1, [{reply, From, ok}])
     end;
+handle_call_(_, {strict_sig_checks, Strict}, From, #data{} = D) ->
```
Comment (uwiger) [inline apps/aechannel/src/aesc_fsm.erl]: It might be good to check the type of `Strict` even here.

File: apps/aechannel/src/aesc_fsm.erl [hunk 2]
```diff
@@ -1306,6 +1384,8 @@ handle_common_event_(cast, {?CHANNEL_CLOSING, ChanId} = Msg, _St, _,
             lager:debug("Channel no longer active: ~p", [Error]),
             close(channel_no_longer_active, D)
     end;
+handle_common_event_({call, From}, {strict_sig_checks, Strict}, _, _, #data{} = D) ->
```
Comment (uwiger) [inline apps/aechannel/src/aesc_fsm.erl]: Perhaps a comment about why a specific clause is needed here, since the next clause seems to handle all calls (except during `Mode == error_all`)?

Also, check that `Strict` is a boolean.

File: apps/aechannel/src/aesc_fsm.erl [hunk 3]
```diff
@@ -1345,7 +1425,7 @@ error_binary(E) when is_atom(E) ->
     atom_to_binary(E, latin1).
 
 
-terminate(Reason, _State, Data) ->
+terminate(Reason, _State, #data{session = _Session} = Data) ->
```
Comment (uwiger) [inline apps/aechannel/src/aesc_fsm.erl]: Why this change? `_Session` doesn't seem to be referred to.

File: apps/aechannel/src/aesc_fsm.erl [hunk 4]
```diff
@@ -1915,7 +2012,14 @@ check_deposit_signed_msg(#{ channel_id := ChanId
                           , data       := TxBin} = Msg,
                           #data{on_chain_id = ChanId} = Data) ->
     SignedTx = aetx_sign:deserialize_from_binary(TxBin),
-    {ok, SignedTx, log(rcv, ?DEP_SIGNED, Msg, Data)}.
+    case check_type_and_verify_signatures(SignedTx, channel_deposit_tx,
+                                          [other_account(Data),
+                                           my_account(Data)],
```
Comment (uwiger) [inline apps/aechannel/src/aesc_fsm.erl]: Not terribly important, but one could imagine a `both_accounts(Data)` helper function. ;-)

File: apps/aechannel/src/aesc_fsm.erl (hunk 3 above)
Comment (velzevur) [reply apps/aechannel/src/aesc_fsm.erl]: Refactoring side effect

File: apps/aechannel/src/aesc_fsm.erl [hunk 5]
```diff
@@ -17,10 +17,13 @@
          close_solo/1,
          shutdown/1,              %% (fsm())
          client_died/1,           %% (fsm())
+         connection_died/1,       %% (fsm())
          inband_msg/3,
          get_state/1,
          get_poi/2]).
 
+-export([strict_sig_checks/2]). %% tests only
```
[the remark this answers is not in the document]
Comment (velzevur) [reply apps/aechannel/src/aesc_fsm.erl]: I'd actually like to play with this in a non-test environment :)
